geodiff/tests/testdata/complex/complex1.json

```diff
@@ -58,7 +58,7 @@
           },
           {
               "column": 2,
-              "new": "\n\t\r\"';,"
+              "new": "\\n\\t\\r\"';,"
```

Comment:
why this changes? was it buggy before?

Reply:
Caret return and some other characters should be escaped. So yes, previously if string had such characters the resulting JSON was incorrect. That's why we got https://github.com/lutraconsulting/qgis-mergin-plugin/issues/378 and added workaround to the QGIS plugin.

Reply:
I think yes. We get it for free just be switching to that JSON library, seems it does all necessary escaping and quoting for us.

Reply:
nice